File: SaturdayQuizWeb/wwwroot/script/QuizMetadata.js
import CalendarDate from "./CalendarDate.js";

export default class QuizMetadata {
    constructor(object) {
        this.id = object.id;
        this.date = new CalendarDate(object.date);
        this.title = object.title;
        this.url = object.url;
        this.source = object.source;
    }
}

File: SaturdayQuizWeb/Models/QuizMetadata.cs
using SaturdayQuizWeb.Extensions;

namespace SaturdayQuizWeb.Models;

public sealed record QuizMetadata
{
    public required string Id { get; init; } = string.Empty;

    public required DateTime Date
    {
        get;
        init => field = value.ToDateUtc();
    }

    public string Title { get; init; } = string.Empty;

    public string Url { get; init; } = string.Empty;

    public string Source { get; init; } = string.Empty;

    public override int GetHashCode() => Url.GetHashCode();

    public bool Equals(QuizMetadata? other) => Url.Equals(other?.Url);
}
